### ISSUE TITLE
0.3.0-alpha.01 should be rejected

### ISSUE THREAD

**Issue by Qwaz** [dtolnay/semver#230]
The version string `0.3.0-alpha.01` is currently accepted by `semver` 0.11.0.

[Playground Link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=14607813d55b51ab6caaca82ef2358bf)

The issue is that `01` is not a valid pre-release identifier. According to [Semantic Versioning 2.0.0 reference](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions), a pre-release identifier can be either a numeric identifier or an alphanumeric identifier. However, the identifier `01` is neither of them. Numeric identifiers cannot start with 0 (EDIT: unless it is a single digit 0), and alphanumeric identifiers must have at least one non-digit character.

This bug led to the publication of a crate with an invalid version on crates.io:
https://crates.io/crates/hxgm30-client

**Comment by Qwaz** [comment]
Another case found on crates.io: https://crates.io/crates/bluetooth_client/0.0.1-001

**Comment by steveklabnik** [comment]
Yep, this is a bug, thanks. We will probably have to add in a compat shim to https://github.com/steveklabnik/semver/blob/master/src/version_req.rs#L323-L332

**Comment by dtolnay** [comment]
This is fixed in 1.0.0.

```rust
fn main() {
    println!("{}", semver::Version::parse("0.3.0-alpha.01").unwrap_err());
}
```

```console
invalid leading zero in pre-release identifier
```
